

# Crotalus adamanteus structural variant
[![Published in MBE](https://img.shields.io/badge/published%20in-MBE-blue)](https://doi.org/10.1093/gigascience/giad116)
[![Data available in the Fisghare](https://img.shields.io/badge/data%20available%20in%20the-figshare-red)](https://figshare.com/projects/Eastern_diamondback_rattlesnake_Crotalus_adamanteus_-_haplotype-resolved_genome_assembly/200614)

This repository contains commands and scripts used in the manuscript "A Segregating Structural Variant Defines Novel Venom
Phenotypes in the Eastern Diamondback Rattlesnake" published in *Molecular Biology and Evolution*.

# Genome Assembly

# Genome annotation
## Repeat annotation
The repeat annotation was performed using [RepeatModeler2](https://github.com/Dfam-consortium/RepeatModeler), to generate a *de novo* species-specific library, and [RepeatMasker](https://github.com/Dfam-consortium/RepeatMasker), to perform the annotation. We complement the species-sepcific TE library using a curated TE library of snakes previosuly published <sup>[Castoe et al., 2013](https://doi.org/10.1073/pnas.1314475110)</sup>.

The pipeline used to perform the repeat annotation is decribed in the following tutorial: https://github.com/pedronachtigall/Repeat-annotation-pipeline

## Gene annotation
The gene annotation was performed using the [funannotate](https://github.com/nextgenusfs/funannotate) pipeline. We followed the commands decribed in the section "NonToxin annotation" of this guide: https://github.com/pedronachtigall/ToxCodAn-Genome/tree/main/Guide

## Toxin annotation

```

```

The toxin genes were manually curated following the section "Checking annotations" of this guide: https://github.com/pedronachtigall/ToxCodAn-Genome/tree/main/Guide#checking-annotations

# Cite

```
ADD bibtext here
```
